hyperspy/_signals/eds.py

```diff
@@ -644,6 +644,10 @@ def get_lines_intensity(self,
                                                            FWHM_MnKa='auto')
             element, line = utils_eds._get_element_and_line(Xray_line)
             img = self.isig[window[0]:window[1]].integrate1D(-1)
+            if np.issubdtype(np.int32, np.integer):
```

Comment:
Well spotted. I meant:

```python

np.issubdtype(img.data, np.integer)

```
The point is to change the dtype only if needed. Would this be ok?